google-api-client/src/main/java/com/google/api/client/googleapis/mtls/MtlsUtils.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.googleapis.mtls;
+
+import com.google.api.client.googleapis.util.Utils;
+import com.google.api.client.json.JsonParser;
+import com.google.api.client.util.Beta;
+import com.google.api.client.util.SecurityUtils;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.util.List;
+
+/**
+ * {@link Beta} <br>
+ * Utilities for mutual TLS.
+ *
+ * @since 1.31
+ */
+@Beta
+public class MtlsUtils {
+  @VisibleForTesting
+  static class DefaultMtlsProvider implements MtlsProvider {
+    private static final String DEFAULT_CONTEXT_AWARE_METADATA_PATH =
+        System.getProperty("user.home") + "/.secureConnect/context_aware_metadata.json";
+
+    /** GOOGLE_API_USE_CLIENT_CERTIFICATE environment variable */
+    public static final String GOOGLE_API_USE_CLIENT_CERTIFICATE =
+        "GOOGLE_API_USE_CLIENT_CERTIFICATE";
+
+    interface EnvironmentProvider {
+      String getenv(String name);
+    }
+
+    static class SystemEnvironmentProvider implements EnvironmentProvider {
+      @Override
+      public String getenv(String name) {
+        return System.getenv(name);
+      }
+    }
+
+    DefaultMtlsProvider() {
+      this(new SystemEnvironmentProvider(), DEFAULT_CONTEXT_AWARE_METADATA_PATH);
+    }
+
+    private EnvironmentProvider envProvider;
+    private String metadataPath;
+
+    @VisibleForTesting
+    DefaultMtlsProvider(EnvironmentProvider envProvider, String metadataPath) {
+      this.envProvider = envProvider;
+      this.metadataPath = metadataPath;
+    }
+
+    @Override
+    public boolean useMtlsClientCertificate() {
+      String useClientCertificate = envProvider.getenv(GOOGLE_API_USE_CLIENT_CERTIFICATE);
+      return "true".equals(useClientCertificate);
+    }
+
+    @Override
+    public String getKeyStorePassword() {
+      return "";
+    }
+
+    @Override
+    public KeyStore getKeyStore() throws IOException, GeneralSecurityException {
+      // Load the cert provider command from the json file.
+      InputStream stream;
```

Comment:
make the try block bigger so you don't have to split the variable declaration and initialization

Reply:
The current way seems fine since it catches exception right after each call so we know which function throws which exception. A bigger try block would basically wrap the whole implementation.

Reply:
One big try block that wraps all the code is exactly what how exceptions in Java were designed to be used. In this method I see no ambiguity about which lines throw which exceptions.

Reply:
I see. Changed to one big try block.